util/types/convert.go

```diff
@@ -198,6 +198,34 @@ func StrToDuration(sc *variable.StatementContext, str string, fsp int) (t Time,
 	return duration.ConvertToTime(mysql.TypeDuration)
 }
 
+// NumberToDuration converts number to Duration.
+func NumberToDuration(number int64, fsp int) (t Time, err error) {
+	if number > TimeMaxValue {
+		// Try parse DATETIME
+		if number >= 10000000000 { // '2001-00-00 00-00-00'
+			if t, err = ParseDatetimeFromNum(number); err == nil {
+				return t, err
```

Comment:
1. return t, nil ?
2. what if `err != nil`

Reply:
if err != nil, it will fallthrough to treat the number as duration, the behavior is ported by MySQL